src/main/java/Juan.java

```diff
@@ -0,0 +1,139 @@
+import CustomExceptions.*; // Import custom exception classes
+import TaskChildren.*; // Import task-related classes like ToDo, Deadline, and Event
+
+import java.util.Scanner; // Import Scanner for user input
+
+public class Juan {
+    // Constant for common error message to improve code readability and reusability
+    private final static String porFavor = "Por Favor?\n";
+
+    // Main entry point for the application
+    public static void main(String[] args) {
+        // Display initial line and welcome message
+        lineMessage();
+        helloMessage();
+        lineMessage();
+
+        // Continue chatting as long as user doesn't exit
+        boolean continueChatting = true;
+        while (continueChatting) {
+            // Chat feature to handle user input
+            continueChatting = chatFeature();
+        }
+
+        // Display goodbye message when the chat ends
+        byeMessage();
+        lineMessage();
+    }
+
+    // Handles user input and executes corresponding actions
+    public static boolean chatFeature() {
+
+        Scanner scanner = new Scanner(System.in); // Initialize the scanner for reading user input
+        String line = scanner.nextLine(); // Read user input
+        lineMessage(); // Print separator line
+
+        // Handle "bye" command to end chat
+        if (line.equals("bye")) {
+            return false; // End the conversation
+        }
+        // Handle "list" command to display the list of tasks
+        else if (line.equals("list")) {
+            Task.printTasksList(); // Print task list from Task class
+        }
+        // Handle "mark" command to mark a task as completed
+        else if (line.startsWith("mark ")) {
```

Comment:
Try to avoid magic literals(e.g. numbers, strings), you can consider using named constants instead. It will be useful especially when the particular literals are used multiple times in the code.
E.g.
```
static final double PI = 3.14236;
static final int MAX_SIZE = 10;
...
return PI;
...
return MAX_SIZE - 1;

```

Reply:
Thanks for the feedback just updated this!